integration-tests/src/test/java/oracle/kubernetes/operator/utils/Domain.java

```diff
@@ -850,6 +850,109 @@ private void createPV() throws Exception {
+    if (!result) {
+      TestUtils.createNewYamlFile(
+          BaseTest.getUserProjectsDir() + "/weblogic-domains/" + domainUid + "/domain.yaml",
+          BaseTest.getUserProjectsDir() + "/weblogic-domains/" + domainUid + "/domain_new.yaml",
+          oldPropertyString,
+          newPropertyString);
+      logger.info(
+          "Done - generate new domain.yaml for "
+              + domainUid
+              + " oldProperty: "
+              + oldPropertyString
+              + " newProperty: "
+              + newPropertyString);
+
+      // kubectl apply the new generated domain yaml file with changed property
+      StringBuffer command = new StringBuffer();
+      command
+          .append("kubectl apply  -f ")
+          .append(
+              BaseTest.getUserProjectsDir()
+                  + "/weblogic-domains/"
+                  + domainUid
+                  + "/domain_new.yaml");
+      logger.info("kubectl execut with command: " + command.toString());
+      TestUtils.exec(command.toString());
+
+      // verify the servers in the domain are being restarted in a sequence
+      verifyAdminServerRestarted();
+      verifyManagedServersRestarted();
+      // make domain.yaml include the new changed property
+      TestUtils.copyFile(
+          BaseTest.getUserProjectsDir() + "/weblogic-domains/" + domainUid + "/domain_new.yaml",
+          BaseTest.getUserProjectsDir() + "/weblogic-domains/" + domainUid + "/domain.yaml");
+    }
+    logger.info("Done - testDomainServerPodRestart");
+  }
+
+  /**
+   * verify that admin server pod gets restarted.
+   *
+   * @throws Exception
+   */
+  public void verifyAdminServerRestarted() throws Exception {
+    logger.info("Checking if admin pod(" + domainUid + "-" + adminServerName + ") is Terminating");
+    TestUtils.checkPodTerminating(domainUid + "-" + adminServerName, domainNS);
+    Thread.sleep(10 * 1000);
+    logger.info("Checking if admin pod(" + domainUid + "-" + adminServerName + ") is Running");
+    TestUtils.checkPodCreated(domainUid + "-" + adminServerName, domainNS);
+    Thread.sleep(10 * 1000);
+  }
+
+  /**
+   * verify that managed server pods get restarted.
+   *
+   * @throws Exception
+   */
+  public void verifyManagedServersRestarted() throws Exception {
+    if (domainMap.get("serverStartPolicy") == null
+        || (domainMap.get("serverStartPolicy") != null
+            && !domainMap.get("serverStartPolicy").toString().trim().equals("ADMIN_ONLY"))) {
+      // check managed server pods
+      for (int i = 1; i <= initialManagedServerReplicas; i++) {
+        logger.info(
+            "Checking if managed pod("
+                + domainUid
+                + "-"
+                + managedServerNameBase
+                + i
+                + ") is Terminating");
+        TestUtils.checkPodTerminating(domainUid + "-" + managedServerNameBase + i, domainNS);
+        Thread.sleep(10 * 1000);
```

Comment:
TestUtils.checkPodCreated() checks in loop. Is the hard sleep needed here?

Reply:
Yes. This is needed.  Between pod Terminating and Running there was a very short period "containerCreating".

Reply:
Hard sleep is removed.

Reply:
Locally all the tests passed.